File: CarCareHub_/CarCareHub.Model/Narudzba.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Text.Json.Serialization;
using System.Threading.Tasks;

namespace CarCareHub.Model
{
    public class Narudzba
    {
        public int NarudzbaId { get; set; }

      //  public int? NarudzbaStavkeId { get; set; }

        public DateTime? DatumNarudzbe { get; set; }

        public DateTime? DatumIsporuke { get; set; }

        public bool? ZavrsenaNarudzba { get; set; }
        public int? KlijentId { get; set; }
      //  public virtual Klijent? Klijent { get; set; }


        public int? AutoservisId { get; set; }
      //  public virtual Autoservis? Autoservis { get; set; }


        public int? ZaposlenikId { get; set; }
      //  public virtual Zaposlenik? Zaposlenik { get; set; }

        public int? PopustId { get; set; }
        public decimal? UkupnaCijenaNarudzbe { get; set; }
        public bool Vidljivo { get; set; }
    }
}

File: CarCareHub_/UI_/flutter_desktop/lib/models/narudzba.dart
import 'package:flutter_mobile/models/autoservis.dart';
import 'package:flutter_mobile/models/klijent.dart';
import 'package:flutter_mobile/models/narudzba_stavke.dart';
import 'package:flutter_mobile/models/zaposlenik.dart';
import 'package:json_annotation/json_annotation.dart';

part 'narudzba.g.dart';

@JsonSerializable()
class Narudzba {
  final int narudzbaId;
  final DateTime? datumNarudzbe;
  final DateTime? datumIsporuke;
  final bool zavrsenaNarudzba;
  final double? ukupnaCijenaNarudzbe;
  bool? vidljivo;
  int? klijentId;
  Klijent? klijent;
  int? autoservisId;
  Autoservis? autoservis;
  int? zaposlenikId;
  Zaposlenik? zaposlenik;
  String? adresa;

  Narudzba(
      {required this.narudzbaId,
      this.vidljivo,
      this.datumNarudzbe,
      this.datumIsporuke,
      this.zavrsenaNarudzba = false,
      this.ukupnaCijenaNarudzbe,
      this.adresa});

  factory Narudzba.fromJson(Map<String, dynamic> json) =>
      _$NarudzbaFromJson(json);

  Map<String, dynamic> toJson() => _$NarudzbaToJson(this);
}
